execution/chainspec/config.go

```diff
@@ -54,6 +55,122 @@ func ReadChainSpec(fileSys fs.FS, filename string) *chain.Config {
+		Config:      ReadChainConfig(chainspecs, "chainspecs/hoodi.json"),
+		Bootnodes:   HoodiBootnodes,
+		Genesis:     HoodiGenesisBlock(),
+		DNSNetwork:  dnsPrefix + "all.hoodi.ethdisco.net",
+	}
+
+	Gnosis = Spec{
+		Name:             networkname.Gnosis,
+		GenesisHash:      common.HexToHash("0x4f1dd23188aab3a76b463e4af801b52b1248ef073c648cbdc4c9333d3da79756"),
+		GenesisStateRoot: common.HexToHash("0x40cf4430ecaa733787d1a65154a3b9efb560c95d9e324a23b97f0609b539133b"),
+		Config:           ReadChainConfig(chainspecs, "chainspecs/gnosis.json"),
+		Bootnodes:        GnosisBootnodes,
+		Genesis:          GnosisGenesisBlock(),
+	}
+
+	Chiado = Spec{
+		Name:             networkname.Chiado,
+		GenesisHash:      common.HexToHash("0xada44fd8d2ecab8b08f256af07ad3e777f17fb434f8f8e678b312f576212ba9a"),
+		GenesisStateRoot: common.HexToHash("0x9ec3eaf4e6188dfbdd6ade76eaa88289b57c63c9a2cde8d35291d5a29e143d31"),
+		Config:           ReadChainConfig(chainspecs, "chainspecs/chiado.json"),
+		Bootnodes:        ChiadoBootnodes,
+		Genesis:          ChiadoGenesisBlock(),
+	}
+
+	Test = Spec{
+		Name:             networkname.Test,
+		GenesisHash:      common.HexToHash("0x6116de25352c93149542e950162c7305f207bbc17b0eb725136b78c80aed79cc"),
+		GenesisStateRoot: empty.RootHash,
+		Config:           chain.TestChainConfig,
+		//Bootnodes:   TestBootnodes,
+		Genesis: TestGenesisBlock(),
+	}
+)
+
+// Filled by init(); mapping of chain genesis hashes to chain specs.
+var supportedChainGenesisHashes = map[common.Hash]Spec{}
+
+// mapping of chain names to chain specs.
+var supportedChains = map[string]Spec{
+	//networkname.Mainnet: Mainnet,
+	//networkname.Holesky: Holesky,
+	//networkname.Sepolia: Sepolia,
+	//networkname.Hoodi:   Hoodi,
+	//networkname.Gnosis:  Gnosis,
+	//networkname.Chiado:  Chiado,
+	//networkname.Test:    Test,
+
+	"all-clique-protocol-changes": {
+		// AllCliqueProtocolChanges contains every protocol change (EIPs) introduced
+		// and accepted by the Ethereum core developers into the Clique consensus.
+		Name:        "all-clique-protocol-changes",
+		GenesisHash: empty.RootHash,
+		Config: &chain.Config{
+			ChainID:               big.NewInt(1337),
+			Consensus:             chain.CliqueConsensus,
+			HomesteadBlock:        big.NewInt(0),
+			TangerineWhistleBlock: big.NewInt(0),
+			SpuriousDragonBlock:   big.NewInt(0),
+			ByzantiumBlock:        big.NewInt(0),
+			ConstantinopleBlock:   big.NewInt(0),
+			PetersburgBlock:       big.NewInt(0),
+			IstanbulBlock:         big.NewInt(0),
+			MuirGlacierBlock:      big.NewInt(0),
+			BerlinBlock:           big.NewInt(0),
+			LondonBlock:           big.NewInt(0),
+			Clique:                &chain.CliqueConfig{Period: 0, Epoch: 30000},
+		},
+		Genesis: &types.Genesis{},
+	},
+}
+
 // Genesis hashes to enforce below configs on.
 var (
 	MainnetGenesisHash = common.HexToHash("0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3")
```

Comment:
Would be better to keep this constants and reuse in `Spec` or define in Spec and use from the type?